src/main/java/bookstore/repository/BookRepository.java

```diff
@@ -1,11 +1,24 @@
 package bookstore.repository;
 
 import bookstore.model.Book;
+import java.util.List;
 import java.util.Optional;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.repository.EntityGraph;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
 import org.springframework.stereotype.Repository;
 
 @Repository
-public interface BookRepository extends JpaRepository<Book, Long> {
+public interface BookRepository extends JpaRepository<Book, Long>,
+        JpaSpecificationExecutor<Book> {
+    @EntityGraph(attributePaths = "categories")
     Optional<Object> getBookById(Long id);
```

Comment:
```suggestion
    Optional<Book> getBookById(Long id);
```
Do you really need a return book with categories? let's check in which methods it is really necessary

Reply:
not fixed Do you really need a return book with categories? let's check in which methods it is really necessary